NETDefault/UnitTestingWithNUnit&Moq/Service.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Linq;
+
+namespace ServiceTests
+{
+    internal class Service
+    {
+        private IGateway _gateway;
+
+        public Service(IGateway gateway)
+        {
+            _gateway = gateway;
+        }
+
+        public object GetDashboardByUserName(string userName)
+        {
+            string dashboard = string.Empty;
+
+            try
+            {
+                var userId = _gateway.GetUserIdByUserName(userName);
+                dashboard += userId;
+
+                var hobbies = _gateway.GetHobbiesByUserId(userId);
+                AddHobbiesToDashboard(hobbies, ref dashboard);
```

Comment:
Please keep the examples simple instead of getting the `ref`s out. You can just put the `foreach` here, the `if` that checks if there are items isn't even needed.